Xamarin.PropertyEditing.Mac/Controls/BindingEditor/BindingEditorWindow.cs

```diff
@@ -0,0 +1,276 @@
+				NSLayoutConstraint.Create (objectSelectorControl, NSLayoutAttribute.Left, NSLayoutRelation.Equal, this.AncestorTypeBox, NSLayoutAttribute.Left, 1f, 5f),
+				NSLayoutConstraint.Create (objectSelectorControl, NSLayoutAttribute.Width, NSLayoutRelation.Equal, this.AncestorTypeBox, NSLayoutAttribute.Width, 1f, -10f),
+				NSLayoutConstraint.Create (objectSelectorControl, NSLayoutAttribute.Height, NSLayoutRelation.Equal, this.AncestorTypeBox, NSLayoutAttribute.Height, 1f, -10f)
+			});
+
+			var longDescription = new UnfocusableTextField {
+				Alignment = NSTextAlignment.Left,
+				TranslatesAutoresizingMaskIntoConstraints = false,
+				StringValue = string.Empty,
+			};
+
+			this.AncestorTypeBox.AddSubview (longDescription);
+
+			this.AncestorTypeBox.AddConstraints (new[] {
+				NSLayoutConstraint.Create (longDescription, NSLayoutAttribute.Top, NSLayoutRelation.Equal, this.AncestorTypeBox, NSLayoutAttribute.Top, 1f, 10f),
+				NSLayoutConstraint.Create (longDescription, NSLayoutAttribute.Left, NSLayoutRelation.Equal, this.AncestorTypeBox, NSLayoutAttribute.Left, 1f, 10f),
+				NSLayoutConstraint.Create (longDescription, NSLayoutAttribute.Width, NSLayoutRelation.Equal, this.AncestorTypeBox, NSLayoutAttribute.Width, 1f, -10f),
+				NSLayoutConstraint.Create (longDescription, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 24),
+			});
+
+			var pathHeader = new HeaderView {
+				Title = Properties.Resources.Path,
+			};
+
+			this.PathBox.AddSubview (pathHeader);
+
+			this.PathBox.AddConstraints (new[] {
+				NSLayoutConstraint.Create (pathHeader, NSLayoutAttribute.Top, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Top, 1f, 0f),
+				NSLayoutConstraint.Create (pathHeader, NSLayoutAttribute.Left, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Left, 1f, 0f),
+				NSLayoutConstraint.Create (pathHeader, NSLayoutAttribute.Width, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Width, 1f, 0f),
+				NSLayoutConstraint.Create (pathHeader, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 40),
+			});
+
+			var pathSelectorControl = new BindingPathSelectorControl (this.viewModel);
+
+			this.PathBox.AddSubview (pathSelectorControl);
+
+			this.PathBox.AddConstraints (new[] {
+				NSLayoutConstraint.Create (pathSelectorControl, NSLayoutAttribute.Top, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Top, 1f, 5f),
+				NSLayoutConstraint.Create (pathSelectorControl, NSLayoutAttribute.Left, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Left, 1f, 5f),
+				NSLayoutConstraint.Create (pathSelectorControl, NSLayoutAttribute.Width, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Width, 1f, -10f),
+				NSLayoutConstraint.Create (pathSelectorControl, NSLayoutAttribute.Height, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Height, 1f, -10f)
+			});
+
+			this.viewModel.PropertyChanged += (sender, e) => {
```

Comment:
You're missing a check for `CanCreateBinding`, so the dialog allows you to create invalid bindings.

Reply:
`PropertyDisplay` can also change.

Reply:
Thanks.